resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -401,7 +403,7 @@ public DnsNameResolver(
                                                                dnsQueryLifecycleObserverFactory) :
                 checkNotNull(dnsQueryLifecycleObserverFactory, "dnsQueryLifecycleObserverFactory");
         this.searchDomains = searchDomains != null ? searchDomains.clone() : DEFAULT_SEARCH_DOMAINS;
-        this.ndots = ndots >= 0 ? ndots : DEFAULT_NDOTS;
+        this.ndots = ndots >= 0 ? ndots : DEFAULT_OPTIONS.ndots();
```

Comment:
@fabienrenaud I wonder if we should better do this as part of the builder ?

Reply:
Doable. 
But I'd need to keep this `ndots >= 0 ? ndots : DEFAULT_NDOTS` (with DEFAULT_NDOTS loaded from resolv.conf) just in case some users invoke the deprecated constructors with `ndots=-1` and expect the value to be loaded from file automatically.
Let me know which way you prefer.

Reply:
never mind... lets keep it as you did.